src/theme/Root.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useEffect } from 'react'
+import { useLocation } from '@docusaurus/router'
+
+import { initializeAnalytics, sendAnalyticsEvent, Trace, user } from '@uniswap/analytics'
+import { CustomUserProperties, EventName, PageName } from '@uniswap/analytics-events'
+import useDocusaurusContext from '@docusaurus/useDocusaurusContext'
+import { getCLS, getFCP, getFID, getLCP, Metric } from 'web-vitals'
+import { getBrowser } from '../utils/browser'
+
+// Placeholder API key. Actual API key used in the proxy server
+const ANALYTICS_DUMMY_KEY = '00000000000000000000000000000000'
+
+function getCurrentPageFromLocation(locationPathname: string): PageName | undefined | string {
+  if (locationPathname === '/') {
+    return PageName.HOME_PAGE
+  }
+  const pathWithoutInitialSlash = locationPathname.slice(1)
+  const pathWithSlashesReplaced = pathWithoutInitialSlash.replace(/\//g, '-')
+  const pageName = pathWithSlashesReplaced.concat('-page')
+  return pageName
+}
+
+// Default implementation, that you can customize
+export default function Root({ children }: React.PropsWithChildren<{ open: boolean }>) {
+  const { pathname } = useLocation()
+  const currentPage = getCurrentPageFromLocation(pathname)
+
+  const { siteConfig } = useDocusaurusContext()
+
+  const proxyUrl = typeof siteConfig.customFields.analytics === 'string' ? siteConfig.customFields.analytics : undefined
+  initializeAnalytics(ANALYTICS_DUMMY_KEY, proxyUrl)
+
+  useEffect(() => {
```

Comment:
Can we add a comment here explaining when this is expected to run for consistency / easy readability?